pandas/tests/reshape/merge/test_join.py

```diff
@@ -956,3 +958,27 @@ def test_join_empty(left_empty, how, exp):
             expected = expected.rename_axis("A")
 
     tm.assert_frame_equal(result, expected)
+
+
+def test_join_multiindex_categorical_output_index_dtype():
+    # GH#50906
+    df1 = DataFrame(
+        {
+            "idx1": Categorical(["a", "a", "a"]),
+            "idx2": Categorical(["a", "a", "b"]),
+            "data": [1, 2, 3],
+        }
+    ).set_index(["idx1", "idx2"])
+
+    df2 = DataFrame(
+        {
+            "idx1": Categorical(["a", "a", "a"]),
+            "idx2": Categorical(["a", "b", "b"]),
+            "data2": [1, 2, 3],
+        }
+    ).set_index(["idx1", "idx2"])
+
+    for how in ["inner", "outer", "left", "right"]:
+        df = df1.join(df2, how=how)
```

Comment:
Could you name this `result` and compare to an `expected` DataFrame. This will compare the types as well

Reply:
Thank you @mroeschke. I corrected the test as you suggested.